astropy/convolution/core.py

```diff
@@ -88,42 +85,66 @@ def center(self):
         """
         return [axes_size // 2 for axes_size in self._array.shape]
 
+    def calculate_normalization(self, mode='integral'):
+        """
+        Calculate the kernel normalization factor.
+
+        Parameters
+        ----------
+        mode : {'integral', 'peak'}
+            One of the following modes:
+                * 'integral' (default)
+                    Kernel is normalized such that its integral = 1.
+                * 'peak'
+                    Kernel is normalized such that its peak = 1.
+        """
+
+        if mode == 'integral':
+            if self._array.sum() == 0:
+                self._normalization = np.inf
+            else:
+                self._normalization = 1. / self._array.sum()
+        elif mode == 'peak':
+            self._normalization = 1. / self._array.max()
+        else:
+            raise ValueError("invalid mode, must be 'integral' or 'peak'")
+        return self._normalization
+
```

Comment:
I had to think about this a little bit again (and remember the intention of this normalization functions...). I think the idea behind the `self._normalization` attribute was, that it should contain the value you have to multiply the kernels `self._array` with, so that it sums to `1`. So `self._normalization` corresponds to `1. / self._array.sum()`, regardless whether the normalization is `peak` or `integral`. The reason for this is, that the astropy convolution algorithms require a normalized kernel (i.e. a kernel that sums to 1). In that sense the bug could be just fixed by setting `self._normalization = 1.` in the  `self.normalize` function, when using the `integral` mode. Is that correct?